app/controllers/concerns/render_condition_concern.rb

```diff
@@ -3,7 +3,13 @@ module RenderConditionConcern
 
   module ClassMethods
     def check_or_render_not_found(callable, **kwargs)
-      before_action(**kwargs) { render_not_found if !callable.call }
+      before_action(**kwargs) do
+        next if callable.call
+        respond_to do |format|
+          format.html { render_not_found }
+          format.json { render_json_not_found }
+        end
```

Comment:
ok so we have the same idea about `respond_to` block, personally I'd see if we can keep one `render_not_found` but at the end of the day this is the same result